src/components/SpaceCreateOsnap.vue

```diff
@@ -0,0 +1,42 @@
+<script setup lang="ts">
```

Comment:
This is the new osnap toggle logic. it looks and works very similarly to the legacy logic, but also includes a warning for if the user has the old and new plugins installed at the same time.